Add PointerLock unadjustedMovement article (#4107)
GoogleChrome/web.dev: src/site/content/en/blog/mouse-acceleration/index.md

> ---

Reviewer: Should we use `web.dev/disable-mouse-acceleration` instead? Would we ever want to talk about mouse acceleration more generally (not just disabling it, but other aspects of it)?

> title: Boost your FPS games on the web

Reviewer: ```suggestion
title: Disable mouse acceleration to provide a better FPS gaming experience
```
Continuing the convo from the Google Doc (thanks for avoiding `headshot`), when in doubt I look for a title that accurately describes the feature we're talking about. `Boost your FPS games on the web` doesn't give me much of a clue about what this page is actually talking about. If game developers search for this use case, I imagine they'll use `disable mouse acceleration` because the FPS gaming community in general probably knows about this tip

was successful.

```js
function requestPointerLockWithUnadjustedMovement() {
  const promise = myTargetElement.requestPointerLock({
    unadjustedMovement: true,
  });

  if (!promise) {
    console.log("disabling mouse acceleration is not supported");
    return;
  }

  return promise
    .then(() => console.log("pointer is locked"))
    .catch((error) => {
      if (error.name === "NotSupportedError") {
        // Some platforms may not support unadjusted movement.
        // You can request again a regular pointer lock.
        return myTargetElement.requestPointerLock();
      }
    });
}
```

It is possible to toggle between accelerated and non-accelerated mouse movement
data without releasing the pointer lock. Simply request the pointer lock again
with the desired option. If that request fails, the original lock will remain
intact and the returned promise will reject. No pointer lock events will fire
for a failed change request.

### Browser support {: #browser-support }

The Pointer Lock API is [well supported across browsers]. However Chromium-based
browsers (e.g. Chrome, Edge, etc.) are the only ones to support disabling

> OS-level adjustment for mouse acceleration at the time of writing.

Reviewer: ```suggestion
OS-level adjustment for mouse acceleration as of October 2020.
See MDN's [Browser compatibility](https://developer.mozilla.org/en-US/docs/Web/API/Pointer_Lock_API#Browser_compatibility)
table for updates.
```
@beaufortfrancois can you create an issue on BCD asking them to add `unadjustedMovement`? https://github.com/mdn/browser-compat-data/issues

> ### Platform support {: #platform-support }

Reviewer: ```suggestion
### Operating system support {: #os-support }
```
Platform seems a bit ambiguous because the web is also a platform, no

Disabling OS-level adjustment for mouse acceleration is supported on Chrome OS,
macOS Catalina 10.15.1, and Windows. Linux will follow.

## Sample  {: #sample }

You can play with the Pointer Lock API by running the [sample] on Glitch. Be
sure to [check out the source code].

{% Glitch { id: 'unadjusted-movement', path: 'script.js' } %}

## Helpful links {: #helpful }

- [Specification PR](https://github.com/w3c/pointerlock/pull/49)
- [GitHub repository](https://github.com/w3c/pointerlock)
- [ChromeStatus entry](https://www.chromestatus.com/feature/5723553087356928)
- [Chrome tracking bug](https://bugs.chromium.org/p/chromium/issues/detail?id=982379)
- [Intent to ship](TODO)
- [Mozilla's position](https://github.com/mozilla/standards-positions/issues/448)
- [WebKit's position](https://lists.webkit.org/pipermail/webkit-dev/2020-October/031473.html)

## Acknowledgements

Thanks to [James Hollyer], [Thomas Steiner], and [Joe Medley] for their reviews
of this article.

> <!-- lint disable definition-case -->

Reviewer: I'm fine with disabling this lint check if you want to go into the lint config file

Author: Yep!
https://github.com/GoogleChrome/web.dev/pull/4115
